CMakeLists.txt

```diff
@@ -1,16 +1,13 @@
 cmake_minimum_required(VERSION 3.5.1)
 include(CheckCCompilerFlag)
 
```

Comment:
Why adding the 'lib' prefix ?
It makes this PR a bit hard to digest as it changes everything.

Curl does not add this extra lib.
https://github.com/curl/curl/blob/master/CMakeLists.txt#L48

Reply:
https://github.com/libuv/libuv/blob/v1.x/CMakeLists.txt#L2

libuv uses libuv ... I guess it doesn't matter much.

Reply:
Because libdeflate is the project name, it's consistent.